src/librustdoc/html/highlight.rs

```diff
@@ -789,6 +789,9 @@ impl<'a> Iterator for TokenIter<'a> {
     }
 }
 
+/// Used to know if a keyword followed by a `!` should never be treated as a macro.
+const KEYWORDS_FOLLOWABLE_BY_VALUE: &[&str] = &["if", "while", "match", "break", "return"];
```

Comment:
Could you include `impl` as well? To fix negative impls: https://doc.rust-lang.org/nightly/src/core/marker.rs.html#1028

<img width="481" height="121" alt="Screenshot 2025-11-09 at 14-22-40 marker rs - source" src="https://github.com/user-attachments/assets/f2220b1a-2790-4490-94b4-25bd86ca5cbd" />

Of course, then the naming `…_BY_VALUE` won't make sense anymore because the thing following the `impl !` is a trait/type.